### PR TITLE
app-arch/rpm: add 4.19.1

### PR DESCRIPTION
**Pull request by pastalian**
- switch from autotools to cmake
- remove USE flags that do not have proper compile options
- remove old patches and workarounds

Closes: https://bugs.gentoo.org/917034

### CODE REVIEW

**Comment by pastalian** [comment]
> Default to [Sequoia](https://sequoia-pgp.org/) crypto as per [roadmap](https://rpm.org/roadmap.html)

https://rpm.org/wiki/Releases/4.19.0

I guess we will need to package rpm-sequoia for this. But for now, I have enabled the deprecated internal parser.

**Comment by pastalian** [comment]
Also, I would like to add myself as a proxy maintainer. Let me know if you don't mind.

**Comment by thesamesam** [comment]
Please do 4.19.1 instead at this point, thanks!

File: profiles/base/package.use.mask [hunk 1]
```diff
@@ -10,6 +10,10 @@
 # app-text/gocr has been masked for removal, bug #824290
 media-gfx/xsane ocr
 
+# Takuya Wakazono <pastalian46@gmail.com> (2023-11-21)
```
Comment (thesamesam) [inline profiles/base/package.use.mask:13]: Given rpm-sequoia needs Rust, let's do this in the wd40 profiles (as well).

**Comment by pastalian** [comment]
Bumped to 4.19.1. The only change needed was to explicitly disable testsuite.

File: profiles/base/package.use.mask (hunk 1 above)
Comment (pastalian) [reply profiles/base/package.use.mask:13]: done

**Comment by thesamesam** [comment]
Many thanks! Some small followups from me in 4cccb538fd36070d2634168006f0f04c59076ad0 and 4d0e37a0837561d09a8f4e44e1f30740efe13846 and I've filed a rekeywording (https://devmanual.gentoo.org/keywording/index.html#keywording-on-upgrades) bug at https://bugs.gentoo.org/919936.

**Pull request closed without merging**